external/llvm-project/mlir/lib/Conversion/XModelToGPU/XModelToGPU.cpp

```diff
@@ -140,6 +169,9 @@ struct LaunchRewritePattern : public OpRewritePattern<async::LaunchOp> {
     if (!kernelPkg.has_value())
       return rw.notifyMatchFailure(op, "no gpu target");
 
+    // Signal that func is already in gpu.async mode
+    caller->setAttr("gpu.is_async", UnitAttr::get(ctx));
```

Comment:
Per comment on the PR itself ... why not fix up gpu-async-region?

Reply:
This is fixed now. Updated gpu-async-region pass to more flexibly support already async'd gpu ops.